elasticdl/python/tests/worker_mnist_test.py

```diff
@@ -0,0 +1,147 @@
+import os
+import unittest
+
+import grpc
+import numpy as np
+import tensorflow as tf
+
+from elasticdl.proto import elasticdl_pb2
+from elasticdl.python.common.constants import GRPC
+from elasticdl.python.common.hash_utils import string_to_id
+from elasticdl.python.common.model_utils import get_model_spec
+from elasticdl.python.ps.parameter_server import ParameterServer
+from elasticdl.python.worker.worker import Worker
+
+
+def random_batch(batch_size):
+    shape = (28, 28)
+    shape = (batch_size,) + shape
+    num_classes = 10
+    images = tf.random.uniform(shape)
+    labels = tf.random.uniform(
+        [batch_size], minval=0, maxval=num_classes, dtype=tf.int32
+    )
+    return images, labels
+
+
+_model_zoo_path = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "../../../model_zoo"
+)
+_test_model_zoo_path = os.path.dirname(os.path.realpath(__file__))
+
+
+class PserverArgs(object):
+    def __init__(
+        self,
+        grads_to_wait=8,
+        lr_staleness_modulation=0,
+        use_async=False,
+        model_zoo=_test_model_zoo_path,
+        model_def="test_module.custom_model",
+        optimizer="optimizer",
+        port=9999,
+        log_level="INFO",
+    ):
+        self.grads_to_wait = grads_to_wait
+        self.lr_staleness_modulation = lr_staleness_modulation
+        self.use_async = use_async
+        self.model_zoo = model_zoo
+        self.model_def = model_def
+        self.optimizer = optimizer
+        self.port = port
+        self.log_level = log_level
+
+
+class WorkerMNISTTest(unittest.TestCase):
+    def setUp(self):
+        ports = [12345, 12346]
+        self._pserver, self._channel = self._create_pserver_and_channel(ports)
+
+    def tearDown(self):
+        for pserver in self._pserver:
+            pserver.server.stop(0)
+
+    def _create_pserver_and_channel(self, ports):
```

Comment:
More than one test files have `_create_pserver_and_channel`. Better to remove it as a function in `test_utils.py`. (This can be done in another PR.)